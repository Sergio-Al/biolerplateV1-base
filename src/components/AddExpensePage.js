import React from "react";

const AddExpensePage = () => <div>This is from my add expense compenent</div>;

export default AddExpensePage;
