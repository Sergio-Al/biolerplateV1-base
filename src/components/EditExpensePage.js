import React from "react";

const EditExpensePage = () => <div>This is from my Edit Page</div>;

export default EditExpensePage;
