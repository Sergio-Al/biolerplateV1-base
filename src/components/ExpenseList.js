import React from "react";
import { connect } from "react-redux";

const ExpenseList = (props) => (
  <div>
    <h1>ExpenseList</h1>
    {props.expenses.length}
    {props.filters.text}
  </div>
);

const mapStateToProps = (state) => {
  return {
    expenses: state.expenses,
    filters: state.filters,
  };
};

export default connect(mapStateToProps)(ExpenseList); // argument in connect(This of we wanna get of the store)(What we want to connect)
