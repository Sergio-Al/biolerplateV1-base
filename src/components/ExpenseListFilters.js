import React from "react";
import { connect } from "react-redux"; // to connect to the store
import { setTextFilter, sortByAmount, sortByDate } from "../actions/filters";

const ExpenseListFilters = (props) => (
  <div>
    <input
      type="text"
      value={props.filters.text}
      onChange={(e) => {
        props.dispatch(setTextFilter(e.target.value)); // we are able to change the value in the filters store
        console.log(e.target.value);
      }}
    />
    <select
      value={props.filters.sortBy}
      onChange={(e) => {
        if (e.target.value === "date") {
          props.dispatch(sortByDate());
        } else if (e.target.value === "amount") {
          props.dispatch(sortByAmount());
        }
      }}
    >
      <option value="date">Date</option>
      <option value="amount">Amount</option>
    </select>
  </div>
);

const mapStateToProps = (state) => ({
  filters: state.filters,
});
export default connect(mapStateToProps)(ExpenseListFilters); // the first argument is optional if we want to access to more store features
