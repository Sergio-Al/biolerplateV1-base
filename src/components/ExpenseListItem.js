import React from "react";
import { Link } from "react-router-dom";

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
  <div>
    <h3>
      Description: {description}
      <Link to={`/edit/${id}`}> Edit</Link>
    </h3>
    <p>
      {amount} - {createdAt}
    </p>
  </div>
);

export default ExpenseListItem;
