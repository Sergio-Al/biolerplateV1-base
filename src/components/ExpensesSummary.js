import React from "react";
import { connect } from "react-redux";
import numeral from "numeral";
import expensesTotal from "../selectors/expenses-total";
import selectExpense from "../selectors/expenses";

export const ExpensesSummary = (props) => (
  <div>
    Viewing {props.expenseCount} totalling: 
    {numeral(props.expensesTotal / 100).format("$0,0.00")}
  </div>
);

const mapStateToProps = (state) => {
  return {
    expenseCount: selectExpense(state.expenses, state.filters).length,
    expensesTotal: expensesTotal(selectExpense(state.expenses, state.filters)),
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
