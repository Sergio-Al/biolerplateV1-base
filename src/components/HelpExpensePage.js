import React from "react";

const HelpExpensePage = () => <div>This is from my Help Page</div>;

export default HelpExpensePage;
