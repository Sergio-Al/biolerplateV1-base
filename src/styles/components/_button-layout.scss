.button-layout {
  background: $light-blue;
  border-style: none;
  color: white;
  display: inline-block;
  font-size: $font-size-large;
  font-weight: 300;
  line-height: 1;
  padding: $s-size;
  text-decoration: none;
}

.button-layout--link {
  background: none;
}

.button-layout--secondary {
  background: $light-gray;
}
