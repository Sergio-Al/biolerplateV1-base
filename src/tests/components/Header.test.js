import React from "react";
import { shallow } from "enzyme";
import ReactShallowRenderer from "react-test-renderer/shallow"; // this will not render child components (use DOM renderer for that)
import { Header } from "../../components/Header";

// react-test-renderer //render components in javascript code
test("should render header correctly", () => {
  const wrapper = shallow(<Header startLogout={() => {}} />);
  expect(wrapper).toMatchSnapshot();

  //   expect(wrapper.find("h1").text()).toBe("Expensify");
  //   const renderer = new ReactShallowRenderer();
  //   renderer.render(<Header />);
  //   expect(renderer.getRenderOutput()).toMatchSnapshot();
});

test("should call startLogout on button click", () => {
  let startLogout = jest.fn();
  const wrapper = shallow(<Header startLogout={startLogout} />);
  wrapper.find("button").simulate('click')
  expect(startLogout).toHaveBeenCalled();
});
