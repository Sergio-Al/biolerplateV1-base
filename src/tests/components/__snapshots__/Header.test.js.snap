// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render header correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <[object Object]
    activeClassName="is-active"
    exact={true}
    to="/dashboard"
  >
    Home
  </[object Object]>
  <[object Object]
    activeClassName="is-active"
    to="/create"
  >
    Create Expense
  </[object Object]>
  <[object Object]
    activeClassName="is-active"
    to="help"
  >
    help
  </[object Object]>
</header>
`;
